tasks-container-update

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+# Update .cockpit-ci/container to the latest tasks container tag automatically
+
+import argparse
+import asyncio
+import logging
+import sys
+from pathlib import Path
+
+import task
+from lib.aio.github import GitHub
+from lib.aio.jsonutil import JsonError, get_nested, get_strv
+
+sys.dont_write_bytecode = True
+
+
+async def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--dry-run', '-n', action='store_true')
+    parser.add_argument('--debug', action='store_true')
+    parser.add_argument('--file', default='.cockpit-ci/container')
+    parser.add_argument('--org', default='cockpit-project')
+    parser.add_argument('--package', default='tasks')
+    args = parser.parse_args()
+
+    if args.debug:
+        logging.basicConfig(level=logging.DEBUG)
+
+    async with GitHub.api(program='tasks-container-update') as api:
+        for version in await api.get_list(f'orgs/{args.org}/packages/container/{args.package}/versions'):
```

Comment:
So is this the bit which needs authentication, which we discussed in https://github.com/cockpit-project/cockpituous/pull/604#issuecomment-2003103469 ? Indeed it would be quite nice to use `skopeo` for this, as it's universal for all registries and doesn't bind us to the GitHub API

Reply:
We don't have/want skopeo in the tasks container, but thanks to your helpful hit I looked under the hood what it's doing — it turns out there's an API to create a 60-seconds duration anonymous access token which you can use the access the other APIs.

I'm in the process of writing a small version of this in aiohttp which we can use.  I want to see if I can possibly get to a place where I can choose the tag based on which one is equal to `latest` — that's not possible with skopeo's output, either.